source/include/StringUtilities.h

```diff
@@ -424,6 +427,42 @@ class DecodedSubString {
     Utf8Char *_decodedStr;
 };
 
+//------------------------------------------------------------
+//! A class for making temporary %-code encoded SubStrings
+class EncodedSubString {
+ public:
+    EncodedSubString() : _encodedStr(null) { }
+    explicit EncodedSubString(const SubString &s, bool encode = true, bool alwaysAlloc = false);
+
+    void Init(const SubString &s, bool encode, bool alwaysAlloc = false);
+    SubString GetSubString() {
+        SubString s;
+        s.AliasAssignLen(_encodedStr, IntIndex(strlen(ConstCStr(_encodedStr))));
+        return s;
+    }
+    Boolean NeedsEncoding(Utf8Char c) {
+        return !SubString::IsNumberChar(c) && !SubString::IsLetterChar(c) && (c != '+' || c != '*' || c != '_' || c != '-' || c != '$');
```

Comment:
Is this based on the JS side? Did we find a spot in Vireo C++ where it's clear which characters need to be encoded for identifiers?